primeqa/mitqa/utils/model_utils/process_row_retriever_output.py

```diff
@@ -0,0 +1,305 @@
+    The function then preprocesses the data by retrieving the top 100 passages for each question and table pair. 
+    It then creates a new dictionary with all of these information as keys and values. The function returns this list of dictionaries.
+    
+    Args:
+        doc_retriever: Retrieve the passages for each row
+        raw_data: Pass the data to be preprocessed
+        qid_scores_dict: Store the scores of each question-id
+        test: Determine whether the data is being processed for training or testing
+    
+    Returns:
+        The processed data
+    """
+    p = qid_scores_dict
+    processed_data = []
+    num = 0
+    den = 0
+    for d in tqdm(raw_data):
+        pi = preprocess_instance(d,test=test)
+        question_str = pi['question']
+        if not test:
+            answer_text = pi['answer-text']
+        q_id = pi['question_id']
+        table_id = pi['table_id']
+        header = pi['table']['header']
+        cri = get_max_score_row(p,q_id)
+        rows =  [pi['table']['data'][cr] for cr in cri]
+        table_rows = []
+        table_row_passages = []
+        table_row_passages_new = []
+        nm, xl = 0, []
+        for r in rows:
+            one_row = {}
+            passage = ""
+            passages = []
+            for r_v,h in zip(r,header):
+                one_row[h] = r_v["cell_value"]
+                passage+= " ".join(r_v['passages'])
+                passages += r_v['passages']
+            table_rows.append(one_row)
+            table_row_passages.append(passage)
+            table_row_passages_new.append(passages)
+        for r,pr,npr in zip(table_rows,table_row_passages,table_row_passages_new):
+            npi={}
+            npi['question_id'] = q_id
+            npi['question'] = question_str
+            npi['table_id'] = table_id
+            npi['table_row'] = r
+            row_values = [v.lower() for v in r.values()]
+            npi['table_passage_row_old'] = pr
+            npi['table_passage_row'] = pr
+
+            if (len(npr)==0):
+                npr = ""
+            elif (len(npr)==1):
+                npr = npr[0]
+            else:
+                npr = " ".join(get_top_k_passages(doc_retriever,npr, question_str, 100, r))
+
+            
+            npi['table_passage_row'] = npr
+            
+            if not test:
+                npi['answer-text'] = answer_text
+                npi['label'] =1
+
+                if answer_text.lower() in npr.lower() or answer_text.lower() in row_values:
+                    npi['label_new'] = 1
+                else:
+                    npi['label_new'] = 0
+                
+                if (npi['label_new']!=npi['label']):
+                    num += 1
+                den += 1
+
+            processed_data.append(npi)
+    print("total", den, "changed", num, len(processed_data))
+    return processed_data
+
+
+if __name__ == "__main__":
```

Comment:
Are you using this block of code for your personal testing? If thats the case, please remove it.